runtime/sema/return_info.go

```diff
@@ -41,6 +41,31 @@ type ReturnInfo struct {
 	// DefinitelyHalted indicates that (the branch of) the function
 	// contains a definite halt (a function call with a Never return type)
 	DefinitelyHalted bool
+	// DefinitelyReturnedOrHalted indicates that (the branch of)
+	// the function contains a definite return statement
+	// or a definite halt (a function call with a Never return type).
+	//
+	// NOTE: this is NOT the same DefinitelyReturned || DefinitelyHalted:
+	// For example, for the following program:
+	//
+	//   if ... {
+	//       return
+	//
+	//       // DefinitelyReturned = true
+	//	     // DefinitelyHalted = false
+	//	     // DefinitelyReturnedOrHalted = true
+	//   } else {
+	//       panic(...)
+	//
+	//       // DefinitelyReturned = false
+	//	     // DefinitelyHalted = true
+	//	     // DefinitelyReturnedOrHalted = true
+	//   }
+	//
+	//   // DefinitelyReturned = false
+	//   // DefinitelyHalted = false
+	//   // DefinitelyReturnedOrHalted = true
```

Comment:
the fact that `DefinitelyReturnedOrHalted != DefinitelyHalted || DefinitelyReturned` is killing me 😅

Should it be renamed to something like `DefinitelyExited`?

Reply:
Yea I like this idea

Reply:
Naming is hard, thank you for the suggestion! "exited" captures the idea well 👍 
Renamed in e26115ca91767fb78ca9913a4cc6137d09d45a4d